kdc/kerberos5.c

```diff
@@ -2588,6 +2572,12 @@ _kdc_tkt_add_if_relevant_ad(krb5_context context,
 	    krb5_set_error_message(context, ret, "add AuthorizationData failed");
 	    return ret;
 	}
+
+	ade = tkt->authorization_data->val[tkt->authorization_data->len-1];
```

Comment:
this always makes me nervous, haven't looked at the full code path but perhaps an assert that `tkt->authorization_data->len != 0` could be nice

Reply:
Hi Like :)
The code just above adds one wrapped AD to the list, but sure we can assert it, I'll revise it.

Reply:
Maybe I'm being needlessly paranoid :) (Given there are probably a few overflow bugs lurking elsewhere as a @abartlet has mentioned)